packages/react-server/core/components/RootElement.js

```diff
@@ -29,8 +36,10 @@ class RootElement extends React.Component {
 
 				// Okay, now we've complained about it
 				// sufficiently, let's go ahead and update.
-				this.props = _.assign({}, this.props, {childProps});
```

Comment:
Took a look into React internals and it appears that setting `this.props = ...` has no effect whatsoever. Thus the subsequent call to `this.forceUpdate()` re-renders the component without any `prop` updates. By using `setState` this will also fix the issue described in https://github.com/redfin/react-server/issues/884